tests/callbacks/test_oom_observer.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+import pathlib
+
+import pytest
+import torch
+from packaging import version
+from torch.utils.data import DataLoader
+
+from composer import State, Trainer
+from composer.callbacks import MemorySnapshot, OOMObserver
+from composer.loggers import LoggerDestination
+from composer.trainer import Trainer
+from tests.common import RandomClassificationDataset, SimpleModel
+
+
+@pytest.mark.skipif(version.parse(torch.__version__) < version.parse('2.1.0'),
+                    reason='OOM Observer requires PyTorch 2.1 or higher')
+def test_oom_observer_warnings_on_cpu_models(device: str):
+
+    # Error if the user sets device=cpu even when cuda is available
+    del device  # unused. always using cpu
+    ob = OOMObserver()
+    Trainer(
+        model=SimpleModel(),
+        callbacks=ob,
+        device='cpu',
+        train_dataloader=DataLoader(RandomClassificationDataset()),
+        max_duration='1ba',
+    )
+    assert ob._enabled is False
```

Comment:
```suggestion
def test_oom_observer_warnings_on_cpu_models():
    ob = OOMObserver()
    Trainer(
        model=SimpleModel(),
        callbacks=ob,
        train_dataloader=DataLoader(RandomClassificationDataset()),
        max_duration='1ba',
    )
    assert ob._enabled is False
```
I think other tests instead add a pytest context and check that it raises the warning it expects?